### ISSUE TITLE
false positive raising-bad-type for value known to be not None

### ISSUE THREAD

**Issue by vlovich** [PyCQA/pylint#2793]
### Steps to reproduce
1. cat test.py
```
class Foo(object):
  def __init__(self):
    self._exc = None

  def fail(self):
    self._exc = Exception("")

  def doit(self):
    if self._exc:
      exc = self._exc
      self._exc = None
      raise exc
```
2. pylint -E test.py
3.

### Current behavior
```
************* Module test
test.py:12:6: E0702: Raising NoneType while only classes or instances are allowed (raising-bad-type)
```

### Expected behavior
pylint should be smart enough not to warn about this.

### pylint --version output
pylint 2.3.1
astroid 2.2.3
Python 3.7.2 (default, Feb 13 2019, 15:31:10) 
[Clang 10.0.0 (clang-1000.11.45.5)]

**Comment by PCManticore** [comment]
Thanks for the report!
